src/pruna/evaluation/metrics/metric_dsg.py

```
@@ -0,0 +1,296 @@
+            
+            # Initialize VQA model
+            if "mplug" in self.vqa_model.lower():
+                from dsg.vqa_utils import mPLUGVQA
+                self._vqa = mPLUGVQA(device=self.device)
+            elif "instructblip" in self.vqa_model.lower():
+                from dsg.vqa_utils import InstructBLIPVQA
+                self._vqa = InstructBLIPVQA(device=self.device)
+            else:
+                raise ValueError(f"Unsupported VQA model: {self.vqa_model}. Supported: mplug-large, instructblip")
+                
+            # Initialize DSG generator
+            self._dsg_generator = DSGGenerator()
+            self._models_initialized = True
+            
+            pruna_logger.info(f"DSG metric initialized with LLM: {self.llm_model}, VQA: {self.vqa_model}")
+            
+        except ImportError as e:
+            error_msg = (
+                "DSG dependencies not found. Please install with: "
+                "'pip install git+https://github.com/j-min/DSG.git' "
+                "and ensure you have the required dependencies (transformers, torch, etc.)"
+            )
+            pruna_logger.error(error_msg)
+            raise ImportError(error_msg) from e
+        except Exception as e:
+            pruna_logger.error(f"Failed to initialize DSG models: {e}")
+            raise
+
+    def _generate_dsg(self, text: str) -> tuple:
+        """Generate DSG (tuples, dependency, and questions) from text."""
+        try:
+            # Generate DSG components using the DSG library
+            id2tuples, id2dependency, id2questions = self._dsg_generator.generate_dsg(
+                text, self._llm, max_questions=self.max_questions
+            )
+            return id2tuples, id2dependency, id2questions
+        except Exception as e:
+            pruna_logger.warning(f"DSG generation failed for text '{text[:50]}...': {e}")
+            # Return empty results on failure
+            return {}, {}, {}
```

do you think it's better to raise the error here and quit already, since there's not a way to meaningfully recover from this?